scripts/setup-macos.sh

```diff
@@ -28,7 +28,7 @@
 set -e # Exit on error.
 set -x # Print commands that are executed.
 
-SCRIPTDIR=$(dirname "$0")
+SCRIPTDIR=$(dirname "${BASH_SOURCE}")
```

Comment:
BTW, I get a linter warning here and the other line:

(warning) Expanding an array without an index only gives the first element.
See https://github.com/koalaman/shellcheck/wiki/SC2128

Reply:
The intent is to get the first element. I will add an index to suppress the warning.